scriptworker/task.py

```diff
@@ -491,8 +514,35 @@ def prepare_to_run_task(context, claim_task):
             raise
 
 
-# kill {{{1
-async def kill(pid, sleep_time=1):
+# max_timeout {{{1
+async def max_timeout(context, proc, timeout):
+    """Kill the task if it takes longer than ``timeout`` seconds.
+
+    Args:
+        context (scriptworker.context.Context): the scriptworker context.
+        proc (subprocess.Process): the subprocess proc. This is compared against
+            context.proc to make sure we're killing the right task.
+        timeout (int): the timeout, in seconds, for the task.
+
+    Raises:
+        ScriptWorkerTaskException, if the task is killed.
+
+    Returns:
+        None, if the task finishes before the timeout.
+
+    """
+    await asyncio.sleep(timeout)
+    if proc is not context.proc:
+        return
+    await kill_task(
+        context,
```

Comment:
Would it make more sense to pass `proc` here, so that we know that we are killing the process that triggered the timeout? I'm not entirely familiar with the control flow here, but the above check suggests this coroutine could outlive a task, in which case there is going to be a race condition of where the check happens and then `kill_task` starts running, and the `context.proc` gets changed under it.

Thinking about it more, I think this could also use a comment about why we aren't just always killing `proc`, even if it is no longer the current process.

Reply:
Sure, we could pass `proc`, esp since we don't use `context` anywhere else in `kill_task`.
And a comment makes sense.